### PR TITLE
hibernate hw

### PR DESCRIPTION
**Pull request by MaVaYuran**

### CODE REVIEW

File: src/main/java/hibernate/HibernateRunner.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    public static void main(String[] args) throws SQLException {
+
+        try(SessionFactory sessionFactory = HibernateUtil.buildSessionFactory();
+            Session session = sessionFactory.openSession()){
+            System.out.println("ok");
+        }
+//        UserDAOImpl userDAO = UserDAOImpl.getInstance();
+//        System.out.println(userDAO.getUser(2));
+//        TicketDAOImpl ticketDAO = TicketDAOImpl.getInstance();
+//        System.out.println(ticketDAO.getTicket(2));
```
Comment (Alexmansar) [inline src/main/java/hibernate/HibernateRunner.java]: don't use commented code

File: src/main/java/hibernate/dao/TicketDAOImpl.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+            transaction = session.beginTransaction();
+            savedId = (Integer) session.save(ticket);
+            transaction.commit();
+
+        } catch (Exception e) {
+            if (transaction != null) transaction.rollback();
+            throw new RuntimeException(e);
+        } finally {
+            if (session != null) session.close();
+        }
```
Comment (Alexmansar) [inline src/main/java/hibernate/dao/TicketDAOImpl.java]: please, use try with resources

File: src/main/java/hibernate/dao/TicketDAOImpl.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+    @Override
+    public Ticket getTicket(int id) {
+        Session session = null;
+        Transaction transaction = null;
+        Ticket ticket;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            ticket = session.get(Ticket.class, id);
+            transaction.commit();
```
Comment (Alexmansar) [inline src/main/java/hibernate/dao/TicketDAOImpl.java]: not needed. You don't change entity

File: src/main/java/hibernate/dao/TicketDAOImpl.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        Transaction transaction = null;
+        Ticket ticket;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            ticket = session.get(Ticket.class, id);
+            if (ticket != null) {
+                ticket.setTicketType(newTicketType);
+                session.update(ticket);
+                transaction.commit();
```
Comment (Alexmansar) [inline src/main/java/hibernate/dao/TicketDAOImpl.java:78]: here its OK, because we update record

**Review by Alexmansar — CHANGES_REQUESTED** [review]
@MaVaYuran In general doof job) But I have some general comments, so please fix them

**Comment by MaVaYuran** [comment]
Now all methods working

File: src/main/java/hibernate/dao/UserDAOImpl.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        }
+    }
+
+    @Override
+    public boolean delete(int id) {
+
+        try (Session session = sessionFactory.openSession()) {
+            User user;
+            Transaction transaction = session.beginTransaction();
+            user = session.get(User.class, id);
```
Comment (Alexmansar) [inline src/main/java/hibernate/dao/UserDAOImpl.java:54]: in general, you can just use
User user = session.get(User.class, id);

File: src/main/java/hibernate/entity/Ticket.java [hunk 6]
```diff
@@ -29,12 +40,12 @@ public void setId(int id) {
...
 
-    public int getUserId() {
-        return userId;
+    public User getUser() {
+        return user;
     }
 
-    public void setUserId(int userId) {
-        this.userId = userId;
+    public void setUserId(User user) {
```
Comment (Alexmansar) [inline src/main/java/hibernate/entity/Ticket.java]: why setUserId?
just setUser

**Review by Alexmansar — APPROVED** [review]
@MaVaYuran Some small comments) please, fix it and merge)

**Comment by MaVaYuran** [comment]
Alexandr good evening, can i merge it

**Pull request merged**